.github/workflows/publish_release.yml

```diff
@@ -28,27 +30,26 @@ jobs:
         registry-url: 'https://registry.npmjs.org'
     - run: npm ci
     - run: npm run build
+
     - name: "@next: publish with commit timestamp (triggered manually)"
       if: contains(github.ref, 'main') && github.event_name == 'workflow_dispatch'
       run: |
         node utils/build/update_canary_version.js --alpha --commit-timestamp
         utils/publish_all_packages.sh --alpha
-      env:
-        NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
     - name: "@next: publish with today's date (triggered automatically)"
-      if: contains(github.ref, 'main') && github.event_name != 'workflow_dispatch'
+      if: contains(github.ref, 'main') && github.event.schedule
       run: |
         node utils/build/update_canary_version.js --alpha --today-date
         utils/publish_all_packages.sh --alpha
-      env:
-        NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
     - name: "@beta: publish with commit timestamp (triggered automatically)"
-      if: contains(github.ref, 'release') && github.event_name != 'workflow_dispatch'
+      if: contains(github.ref, 'release') && github.event_name == 'push'
       run: |
         node utils/build/update_canary_version.js --beta --commit-timestamp
         utils/publish_all_packages.sh --beta
-      env:
-        NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+    - name: "publish release to NPM"
+      if: contains(github.ref, 'release') && github.event_name == 'release'
+      run: utils/publish_all_packages.sh --release
+
     - name: Azure Login
```

Comment:
Should we move build&publish driver to a separate job? That's assuming I can restart a separate job from the workflow, in case it fails for some reason.

Reply:
We could, but I don't like to repeat the build/versioning logic and if the driver is published in a separate job it would need to read artifacts from the previous one. I'd keep it in one job for now as we haven't seen problems with az uploads.